lib/ansible/modules/files/iso_extract.py

```diff
@@ -69,63 +87,106 @@
 #
 '''
 
-import os
+import os.path
 import shutil
 import tempfile
 
+try:  # python 3.3+
+    from shlex import quote
+except ImportError:  # older python
+    from pipes import quote
+
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.pycompat24 import get_exception
 
 def main():
     module = AnsibleModule(
-        argument_spec = dict(
-            image = dict(required=True, type='path', aliases=['path', 'src']),
-            dest = dict(required=True, type='path'),
-            files = dict(required=True, type='list'),
+        argument_spec=dict(
+            image=dict(type='path', required=True, aliases=['path', 'src']),
+            dest=dict(type='path', required=True),
+            files=dict(type='list', required=True),
+            force=dict(type='bool', default=True, aliases=['thirsty']),
+            executable=dict(type='path', default='7z'),
         ),
-        supports_check_mode = True,
+        supports_check_mode=True,
     )
     image = module.params['image']
     dest = module.params['dest']
     files = module.params['files']
+    force = module.params['force']
+    executable = module.params['executable']
+
+    result = dict(
+        changed=False,
+        dest=dest,
+        image=image,
+    )
 
-    changed = False
+    binary = module.get_bin_path(executable, None)
+
+    if not binary:
+        module.fail_json(msg='Executable "%s" is not found on the system' % executable)
 
     if not os.path.exists(dest):
         module.fail_json(msg='Directory "%s" does not exist' % dest)
 
     if not os.path.exists(os.path.dirname(image)):
         module.fail_json(msg='ISO image "%s" does not exist' % image)
 
+    result['files'] = []
+    extract_files = list(files)
```

Comment:
I'm probably just having a brain-fail due to the virus I'm suffering from, but isn't `files` already a list?